pkg/context/notice.go

```diff
@@ -0,0 +1,63 @@
+package context
+
+import (
+	"os"
+	"path"
+	"strings"
+
+	"github.com/Unknwon/com"
+	"github.com/gogs/gogs/pkg/setting"
+	"github.com/gogs/gogs/pkg/tool"
+	log "gopkg.in/clog.v1"
+)
+
+// readServerNotice checks if a notice file exists and loads the message to display
+// on all pages.
+func readServerNotice() map[string]interface{} {
+	fileloc := path.Join(setting.CustomPath, "notice")
+
+	// Limit size to prevent very large messages from breaking pages
+	var maxlen int64 = 1024
+
+	if !com.IsExist(fileloc) {
```

Comment:
It would be better if we check existence of the file before define variable `maxSize`, because `maxSize` is not used until we `.Stat()` it.